TombLib/TombLib/LevelData/Compilers/TombEngine/TombEngineSpriteAllocator.cs

```diff
@@ -0,0 +1,290 @@
+
+            int padding = 8;
+
+            //Sort textures for packing order...
+            int[] usedTexturesProcessingOrder = new int[_usedTextures.Count];
+            for (int i = 0; i < _usedTextures.Count; ++i)
+                usedTexturesProcessingOrder[i] = i;
+            Array.Sort(usedTexturesProcessingOrder, new TextureComparer { _usedTextures = _usedTextures });
+
+            bool done;
+            int atlasWidth = 256;
+            int atlasHeight = 256;
+
+            do
+            {
+                RectPacker texPacker = new RectPackerTree(new VectorInt2(atlasWidth, atlasHeight));
+
+                done = true;
+
+                foreach (int UsedTextureIndex in usedTexturesProcessingOrder)
+                {
+                    TextureView usedTexture = _usedTextures[UsedTextureIndex];
+
+                    // Get the size of the quad surrounding texture area, typically should be the texture area itself
+                    int w = (int)(usedTexture.Area.Width);
+                    int h = (int)(usedTexture.Area.Height);
+
+                    // Calculate adaptive padding at all sides
+                    int tP = padding;
+                    int bP = padding;
+                    int lP = padding;
+                    int rP = padding;
+
+                    int horizontalPaddingSpace = 4096 - w;
+                    int verticalPaddingSpace = 4096 - h;
+
+                    // If hor/ver padding won't fully fit, get existing space and calculate padding out of it
+                    if (verticalPaddingSpace < tP + bP)
+                    {
+                        tP = verticalPaddingSpace / 2;
+                        bP = verticalPaddingSpace - tP;
+                    }
+
+                    if (horizontalPaddingSpace < padding * 2)
+                    {
+                        lP = horizontalPaddingSpace / 2;
+                        rP = horizontalPaddingSpace - lP;
+                    }
+
+                    w += lP + rP;
+                    h += tP + bP;
+
+                    // Pack texture
+                    VectorInt2? pos = texPacker.TryAdd(new VectorInt2(w, h));
+                    if (pos.HasValue)
+                    {
+                        usedTexture.Position = pos.Value;
+                        _usedTextures[UsedTextureIndex] = usedTexture;  
+                    }
+                    else
+                    {
+                        done = false;
+                        atlasWidth += 256;
+                        atlasHeight += 256;
+                        break;
+                    }
+                }
+            } while (!done);
+
+            ImageC resultingTexture = ImageC.CreateNew(atlasWidth, atlasHeight);
+
+            List<ImageC> resultingTextures = new List<ImageC>();
+            _usedTexturePackInfos = new Result[_usedTextures.Count];
+
+            for (int i=0;i<_usedTextures.Count;i++)
```

Comment:
Please uncondense this line with spaces.

Reply:
Fixed :-)